landoapi/patches.py

```diff
@@ -88,14 +93,14 @@ def upload(
 
 
 def download(
-    revision_id,
-    diff_id,
-    s3_bucket,
+    revision_id: int,
+    diff_id: int,
+    s3_bucket: str,
     *,
-    aws_access_key,
-    aws_secret_key,
-    endpoint_url=None
-):
+    aws_access_key: str,
+    aws_secret_key: str,
+    endpoint_url: Optional[str] = None,
```

Comment:
nit: Could use `|` here to be more consistent (and I personally prefer that but take it or leave it).
```suggestion
    endpoint_url: str | None = None,
```

Reply:
I use `Optional` for the most part because of `Option` in Rust and other languages. 

I like this syntax in favor of `Union` (ie below for `bool | str`), but for `something | None` I generally prefer Optional. :)